src/utils/api/api.interface.ts

```diff
@@ -165,13 +171,24 @@ export interface ProfileContent {
   introMessage: string;
 }
 
-export interface BasicCalloutData {
+export interface GetBasicCalloutData {
```

Comment:
I'd like us to discuss this prefixing again, actually. Reading the implementations of this interface, I now feel the `Get-` prefix is an indirection:

`fn = query: GetCalloutsQuery => BasicCalloutData` makes more sense to me.

Also, if this interface is "basic", what is its non-basic version?

Reply:
Yeah definitely up for chatting about this as I agree it does seem a bit pointless. I added a `Get` prefix so that both `Get` and `Update` could derive from the same base class (non-prefixed). 

```typescript
interface Item {
  id: string;
  // other shared properties
}

interface GetItem extends Item {
  created: Date;
  // other read only properties
}

interface UpdateItem extends Item {
  // write only properties??
}
```

Maybe a better route would be `Item` => `BaseItem`, `GetItem` => `Item`?

On the `Basic` vs `More`, this is to reflect that you get more data back from a request to a specific resource than when you load a list of resources.
e.g.
`GET /api/1.0/callout` loads a list of callouts with basic info
`GET /api/1.0/callout/specific-callout` loads a specific callout with more info
Again open to discussing what the right convention is

Reply:
Yeah, I think I understand your motivation behind separating, extending and prefixing the types.

For now, my point is solely about terminology. Function names which include a verb lead you to guess what they do, and so I wonder if this signpost belongs on the types as well.

In this instancem this is the signature of a function using this type:

```ts
fn fetchCallouts = query?: GetCalloutsQuery => Promise<Paginated<GetBasicCalloutData>>
```

`Get-` prefix feels right on the function parameter ; but dissonant on the return.